src/details/ArborX_DetailsUtils.hpp

```diff
@@ -141,7 +141,8 @@ lastElement(Kokkos::View<T, P...> const &v)
   ARBORX_ASSERT(n > 0);
   auto v_subview = Kokkos::subview(v, n - 1);
   auto v_host = Kokkos::create_mirror_view(v_subview);
-  Kokkos::deep_copy(v_host, v_subview);
+  Kokkos::deep_copy(typename Kokkos::View<T, P...>::execution_space{}, v_host,
+                    v_subview);
```

Comment:
Same comment here.  There is no point passing a default constructed execution space and I am unsure at this time whether it would make sense at all to forward the execution space all the way here.

Reply:
It's all about the fences, isn't it? Anyway, I removed it for now.